tests/appsec/test_automated_user_and_session_tracking.py

```diff
@@ -0,0 +1,187 @@
+
+    def setup_user_tracking_sdk_overwrite(self):
+        self.r_login = weblog.post("/login?auth=local&sdk_event=success&sdk_user=sdkUser", data=login_data(context))
+
+    def test_user_tracking_sdk_overwrite(self):
+        assert self.r_login.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_login):
+            meta = span.get("meta", {})
+            assert meta["usr.id"] == "sdkUser"
+            assert meta["_dd.appsec.usr.id"] == "social-security-id"
+            assert meta["_dd.appsec.user.collection_mode"] == "sdk"
+
+
+CONFIG_ENABLED = (
+    "datadog/2/ASM_FEATURES/asm_features_activation/config",
+    {"asm": {"enabled": True}},
+)
+
+BLOCK_USER = (
+    "datadog/2/ASM_DD/rules/config",
+    {
+        "version": "2.1",
+        "metadata": {"rules_version": "1.2.6"},
+        "rules": [
+            {
+                "id": "block-users",
+                "name": "Block User Addresses",
+                "tags": {"type": "block_user", "category": "security_response"},
+                "conditions": [
+                    {
+                        "parameters": {"inputs": [{"address": "usr.id"}], "data": "blocked_users"},
+                        "operator": "exact_match",
+                    }
+                ],
+                "transformers": [],
+                "on_match": ["block"],
+            }
+        ],
+        "rules_data": [
+            {
+                "id": "blocked_users",
+                "type": "data_with_expiration",
+                "data": [{"value": "social-security-id", "expiration": 0}, {"value": "sdkUser", "expiration": 0}],
+            },
+        ],
+    },
+)
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@features.user_monitoring
+@scenarios.appsec_runtime_activation
+class Test_Automated_User_Blocking:
+    def setup_user_blocking_auto(self):
+        rc.rc_state.reset().apply()
+
+        self.config_state_1 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.r_login = weblog.post("/login?auth=local", data=login_data(context))
+
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER).apply()
+        self.r_home_blocked = weblog.get("/", cookies=self.r_login.cookies,)
+
+    def test_user_blocking_auto(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.r_login.status_code == 200
+
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_home_blocked, rule="block-users")
+        assert self.r_home_blocked.status_code == 403
+
+    def setup_user_blocking_sdk(self):
+        rc.rc_state.reset().apply()
+
+        self.config_state_1 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER).apply()
+        self.r_login_blocked = weblog.post(
```

Comment:
should we also have a test for a non blocked user id?
for example one where `sdk_user=sdkGoodUser` and we check it's not blocked, and this way, we can also confirm that auto user id  was well overridden by the sdk (as otherwise it would block on `social-security-id`

Reply:
Add this kind of test but the other way around with the user `testuuid`:

- First request with `testuuid` should be OK as it is not blocked
- Second request uses `sdkUser` overriding the auto instrumentation effectively blocking it